pkg/load/load.go

```diff
@@ -121,10 +122,57 @@ func fromPath(path string) (filenameToConfig, error) {
 	return configs, utilerrors.NewAggregate([]error{err, errGroup.Wait()})
 }
 
-func Config(path, unresolvedPath, registryPath string, info *ResolverInfo) (*api.ReleaseBuildConfiguration, error) {
+var testPathRegex = regexp.MustCompile(`(?P<org>[^/]+)/(?P<repo>[^@]+)@(?P<branch>[^:]+):(?P<test>.+)`)
+
+func parseCiopTestPath(path string) (ciopTestPath, error) {
```

Comment:
Can we just pass JSON please to the flag

Reply:
you mean `--inject-test='{"org": "o","repo": "r","branch": "b","test": "t"}`? That's super ugly and terrible to use...

I know you hate parsing information from strings, but string inputs are comfortable and readable, and regex parsing is not even that ugly.

Reply:
I don't expect humans to pass this, but I expect humans (mostly DPTP) to occasionally eyeball the calls with the params, and I think it is useful if they are readable.

I don't find regex parsing here that complex or fragile - it's not using any complicated patterns that basically only deals with separators) and it could be even read by some `scanf` (using regex with named groups instead of scanf is probably my Python-influenced preference). Of course I'd add tests.

How strong are you on this? I agree that parsing JSON will make the code simpler, but at the expense of human observability, which is IMO rarely a good trade.

Reply:
Humans should not read the flags we generate on the fly inside of a ProwJob YAML either - let's make the `ci-operator` log output clear when it's doing this on resolution?